pkg/atlas/meta.go

```diff
@@ -20,9 +25,55 @@ type InstanceSize struct {
 // GetProvider will find a provider by name using the private API.
 // GET /cloudProviders/{NAME}/options
 func (c *HTTPClient) GetProvider(name string) (*Provider, error) {
-	path := fmt.Sprintf("cloudProviders/%s/options", name)
-
 	var provider Provider
-	err := c.requestPrivate(http.MethodGet, path, nil, &provider)
+
+	// First check to see if any of the providers have been set by the administrator of the cluster locally
+	provider, isSet, err := getLocalProvider(name, c.ProvidersConfig)
+	if err != nil {
+		return &provider, err
+	}
+
+	// Provider exists in local file
+	if isSet {
+		return &provider, err
+	} else if name == TENANT { // Provider TENANT isn't set locally, but the user requires it. Return the harcoded version of it
+		return &Provider{
+			Name: TENANT,
+			InstanceSizes: map[string]InstanceSize{
+				"M2": {Name: "M2"},
+				"M5": {Name: "M5"},
+			},
+		}, err
+	}
+
+	// Otherwise we make a request to the private atlas API when it's not set
+	path := fmt.Sprintf("cloudProviders/%s/options", name)
+	err = c.requestPrivate(http.MethodGet, path, nil, &provider)
 	return &provider, err
 }
+
+// getLocalProvider will setup all of the available providors together with their plans/sizes.
+// Users may want to set their local provider, and combine that with a remote provider that isn't set locally.
+func getLocalProvider(name string, pathToFile string) (provider Provider, isSet bool, err error) {
```

Comment:
We need to check against the situation when the env variable is not set (`pathToFile` is empty) and return `isSet=false` in this case